soc/arm/common/cortex_m/arm_mpu_mem_cfg.h

```diff
@@ -62,6 +62,48 @@
 #error "Unsupported sram size configuration"
 #endif
 
+#define MPU_REGION_SIZE_32		REGION_32B
+#define MPU_REGION_SIZE_64		REGION_64B
+#define MPU_REGION_SIZE_128		REGION_128B
+#define MPU_REGION_SIZE_256		REGION_256B
+#define MPU_REGION_SIZE_512		REGION_512B
+#define MPU_REGION_SIZE_1024		REGION_1K
+#define MPU_REGION_SIZE_2048		REGION_2K
+#define MPU_REGION_SIZE_4096		REGION_4K
+#define MPU_REGION_SIZE_8192		REGION_8K
+#define MPU_REGION_SIZE_16384		REGION_16K
+#define MPU_REGION_SIZE_32768		REGION_32K
+#define MPU_REGION_SIZE_65536		REGION_64K
+#define MPU_REGION_SIZE_131072		REGION_128K
+#define MPU_REGION_SIZE_262144		REGION_256K
+#define MPU_REGION_SIZE_524288		REGION_512K
+#define MPU_REGION_SIZE_1048576		REGION_1M
+#define MPU_REGION_SIZE_2097152		REGION_2M
+#define MPU_REGION_SIZE_4194304		REGION_4M
+#define MPU_REGION_SIZE_8388608		REGION_8M
+#define MPU_REGION_SIZE_16777216	REGION_16M
+#define MPU_REGION_SIZE_33554432	REGION_32M
+#define MPU_REGION_SIZE_67108864	REGION_64M
+#define MPU_REGION_SIZE_134217728	REGION_128M
+#define MPU_REGION_SIZE_268435456	REGION_256M
+#define MPU_REGION_SIZE_536870912	REGION_512M
+
+#define MPU_REGION_SIZE(x)		MPU_REGION_SIZE_ ## x
+
+#define BUILD_MPU_REGION(p_name, p_base, p_size, p_attr)	\
+	{ .name = p_name,					\
+	  .base = p_base,					\
+	  .attr = p_attr(MPU_REGION_SIZE(p_size)),		\
+	},
+
+#else
+
+#define BUILD_MPU_REGION(p_name, p_base, p_size, p_attr)	\
```

Comment:
nit: `ARM_MPU_REGION_INIT`? If comma is added by the DT macro, it has the potential to be used as a standalone initializer (not sure if useful, just commenting)

Reply:
I like it